Add sotw banner to front page pattern

tellyworth opened the pull request:
I've added this above the survey banner; is that correct?

### Screenshots

<!-- If your change has a visual component, add a screenshot here. -->

<img width="1512" alt="Screen Shot 2022-12-08 at 6 31 06 pm" src="https://user-images.githubusercontent.com/7200686/206386066-b7949917-f976-47f5-94eb-24147ba6cacf.png">

<img width="446" alt="Screen Shot 2022-12-08 at 6 31 22 pm" src="https://user-images.githubusercontent.com/7200686/206386075-6b1831c3-542a-4ff0-a5b3-0de7dcd997e9.png">

<!-- If you can, add the appropriate [Component] label(s). -->

tellyworth committed:
Add sotw banner to front page pattern

I've added this above the survey banner; is that correct?

thetinyl commented:
Thanks @tellyworth! To clarify and confirm: the State of the Word banner will be replacing the survey banner entirely. The survey banner will be swapped back in a week or so after the State of the Word event.

iandunn commented on source/wp-content/themes/wporg-main-2022/patterns/home.php:
```diff
@@ -6,6 +6,28 @@
  */
 
 ?>
+<!-- wp:group {"align":"full","style":{"color":{"background":"#020c3a"}},"layout":{"type":"constrained"}} -->
```
It could be convenient to add an `if ( time() < strtotime( '2022-12-22' ) )` condition around this (and the reverse condition for removing the SotW content).

That way we they'll be added/removed at the right times automatically, rather than us having to manually make another pr/deploy.

tellyworth replied:
Good suggestion @iandunn, thanks. I'm going to deploy a change with a php conditional manually included. It's a departure from the usual process, so I'm curious to hear what @ryelle thinks when she's back. It can be overwritten by a different change when necessary.

Long term, it would be useful to have conditional blocks so we can do this kind of thing without needing php code.

ryelle replied:
I think we can do that for specific cases (like this, when it's very important that something be shown/hidden at a certain time), but generally this file shouldn't be edited manually — it's generated from the live page content via `yarn build:patterns`. Running that again if we needed to update the copy, for example, would blow away this conditional.

> Long term, it would be useful to have conditional blocks so we can do this kind of thing without needing php code.

Yeah, an editor-first solution would be better.

tellyworth committed:
Make banner conditional

This is a departure from the usual content-to-pattern workflow because I manually edited the template. Happy to undo that if it turns out to be a terrible idea.
